app/composables/useAuth.ts

```diff
@@ -0,0 +1,95 @@
+import { ref } from "vue";
+import { useRouter } from "vue-router";
+
+const me = ref(null); // This variable stores the current logged in user data
+const error = ref(null);
+
+export function useAuth() {
+  const supabase = useSupabaseClient();
+  const router = useRouter();
+
+  async function signup(
+    email: string,
+    password: string,
+    username: string,
+    first_name: string,
+    last_name: string
+  ) {
+    error.value = null;
+    try {
+      const { data, error: signUpError } = await supabase.auth.signUp({
+        email,
+        password,
+      });
+
+      if (signUpError) throw signUpError;
+
+      const newUser = {
+        user_id: data.user?.id,
+        username,
+        first_name,
+        last_name,
+        email,
+        created_at: new Date(),
+        updated_at: new Date(),
+      };
+
+      const { data: insertedData, error: insertError } = await supabase
+        .from("users")
+        .insert(newUser);
```

Comment:
Can you check out the Typescript complaint on this line?

Reply:
Calling API resolved this typescript complaint